js/meshopt_clusterizer.js

```diff
@@ -0,0 +1,443 @@
+
+	const wasmpack = new Uint8Array([
+		32, 0, 65, 2, 1, 106, 34, 33, 3, 128, 11, 4, 13, 64, 6, 253, 10, 7, 15, 116, 127, 5, 8, 12, 40, 16, 19, 54, 20, 9, 27, 255, 113, 17, 42, 67,
+		24, 23, 146, 148, 18, 14, 22, 45, 70, 69, 56, 114, 101, 21, 25, 63, 75, 136, 108, 28, 118, 29, 73, 115,
+	]);
+
+	if (typeof WebAssembly !== 'object') {
+		return {
+			supported: false,
+		};
+	}
+
+	var instance;
+
+	// for whatever reason calls to meshopt_computeClusterBounds invalidate the wasm function table for subsequent calls to meshopt_buildMeshlets(Scan) or meshopt_optimizeMeshlet
+	// workaround: keep a copy of the initial state of the wasm memory and reset the instance's memory before any call (except for computing bounds) if the last call was to meshopt_computeClusterBounds
+	let initialMemoryState;
```

Comment:
This is currently not validated but the non-module files should not use ES6 features. Probably fine to use in test code.

Reply:
Then I'll make all `let`s and `const`s and the iterator stuff.
Maybe I'm missing something but aren't typed arrays (like `Uint8Array` and so on) an ES6 feature that's used in all JS bindings already?
At least they are not mentioned in the [ES5 spec](https://262.ecma-international.org/5.1/) but they are in the [ES6 spec](https://262.ecma-international.org/6.0/#sec-typedarray-objects).

Reply:
removed es6 features.
`buildMeshlets` now returns an array of meshlets (each has a vertices and triangles array, as well as a reference to the packed buffers that its vertices and triangles arrays are subarrays of)
`computeMeshletBounds` now takes either a single meshlet or a meshlet array. If an array with more than one meshlet is passed in, an array of bounds is returned, otherwise a single bounds object is returned. Bounds are now raw objects

Reply:
fixed remaining es5 issues & added clusterizer to es5 validation workflow